packages/devtools_app/lib/src/debugger/file_search.dart

```diff
@@ -124,72 +123,187 @@ class FileSearchFieldState extends State<FileSearchField>
+    return script.uri.caseInsensitiveContains(query);
+  }
+
+  AutoCompleteMatch createExactFullPathAutoCompleteMatch(ScriptRef script) {
+    if (isEmpty) return AutoCompleteMatch(script.uri);
+
+    final matchedSegments = _findExactSegments(script.uri);
+    return AutoCompleteMatch(script.uri, matchedSegments: matchedSegments);
+  }
+
+  bool isFuzzyMatch(ScriptRef script) {
+    if (isEmpty) return false;
+
+    final fileName = _getFileName(script.uri);
+    return fileName.caseInsensitiveFuzzyMatch(query);
+  }
+
+  AutoCompleteMatch createFuzzyMatchAutoCompleteMatch(ScriptRef script) {
+    if (isEmpty) return AutoCompleteMatch(script.uri);
+
+    final fileName = _getFileName(script.uri);
+    final fileNameIdx = script.uri.lastIndexOf(fileName);
+    final matchedSegments = _findFuzzySegments(fileName)
+        .map((range) =>
+            Range(range.begin + fileNameIdx, range.end + fileNameIdx))
+        .toList();
+
+    return AutoCompleteMatch(script.uri, matchedSegments: matchedSegments);
+  }
+
+  List<Range> _findExactSegments(String file) {
     final autoCompleteResultSegments = <Range>[];
-    if (match.contains(query)) {
-      final start = match.indexOf(query);
-      final end = start + query.length;
-      autoCompleteResultSegments.add(Range(start, end));
-    } else {
-      // For fuzzy-matches, only match on the file name:
-      final fileName = match.split('/').last;
-      var queryIndex = 0;
-      for (int matchIndex = match.indexOf(fileName);
-          matchIndex < match.length;
-          matchIndex++) {
-        if (queryIndex == query.length) break;
-        if (match[matchIndex] == query[queryIndex]) {
-          final start = matchIndex;
-          final end = matchIndex + 1;
-          autoCompleteResultSegments.add(Range(start, end));
-          queryIndex++;
-        }
+    final start = file.indexOf(query);
+    final end = start + query.length;
+    autoCompleteResultSegments.add(Range(start, end));
+    return autoCompleteResultSegments;
+  }
+
+  List<Range> _findFuzzySegments(String file) {
+    final autoCompleteResultSegments = <Range>[];
+    var queryIndex = 0;
+    for (int matchIndex = 0; matchIndex < file.length; matchIndex++) {
+      if (queryIndex == query.length) break;
+      if (file[matchIndex] == query[queryIndex]) {
+        final start = matchIndex;
+        final end = matchIndex + 1;
+        autoCompleteResultSegments.add(Range(start, end));
+        queryIndex++;
       }
     }
+    return autoCompleteResultSegments;
+  }
 
-    return AutoCompleteMatch(
-      match,
-      matchedSegments: autoCompleteResultSegments,
-    );
+  String _getFileName(String fullPath) {
+    return _fileNamesCache[fullPath] ??= fullPath.split('/').last;
   }
+}
 
-  List<ScriptRef> _findMatches(
-    String query,
-    List<ScriptRef> scriptRefs,
-  ) {
+class FileSearchResults {
+  FileSearchResults.emptyQuery(this.allScripts) : query = FileQuery.empty();
+
+  FileSearchResults.withQuery({
```

Comment:
nit use factory constructors instead. You should be able to remove the "// ignore: ..." line by using a factory constructor since asserts are allowed in the body of a factory constructor

Reply:
Oh perfect, thanks!